query/stdlib/influxdata/influxdb/operators.go

```diff
@@ -80,14 +80,31 @@ func (s *ReadRangePhysSpec) Copy() plan.ProcedureSpec {
 }
 
 func (s *ReadRangePhysSpec) PostPhysicalValidate(id plan.NodeID) error {
-	if s.Bounds.Start.IsZero() && s.Bounds.Stop.IsZero() {
+	if s.Bounds.IsEmpty() {
+		// The only way this procedure spec can be instantiated
+		// is if a range was successfully pushed down to storage.
+		// If said range is empty, invalidate the query plan as
+		// as the planner has determined that no storage scan
+		// will take place.
 		var bucket string
 		if len(s.Bucket) > 0 {
 			bucket = s.Bucket
 		} else {
 			bucket = s.BucketID
 		}
-		return fmt.Errorf(`%s: results from "%s" must be bounded`, id, bucket)
+		beg, err := s.Bounds.Start.MarshalText()
+		if err != nil {
+			return err
+		}
+		end, err := s.Bounds.Stop.MarshalText()
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("empty time range queried from bucket %s; start: %s, stop: %s",
```

Comment:
Why is this an error?  Couldn't we fulfill this request, even if it's likely a mistake by the user?

Reply:
If storage cannot handle the empty request, that's a different story.

Reply:
Storage can handle an empty time range, but a query specifying an empty range is most likely an unintended user error. In that case maybe it's best to highlight it vs executing the query and putting some load on the storage engine just to return no data?

Reply:
I can imagine uses for querying an empty range.  Maybe the user just wants to see if their query _would_ execute okay.  In general I don't like to second-guess the user's intention.

One could imagine creating a no-op source that generates no data in the future, as an optimization for things like `filter(fn: (r) => false)` as well as the empty range.

I know erroring on an empty range is the current behavior, so leaving it as-is would be okay too.